geth/peers/peerpool_test.go

```diff
@@ -133,3 +134,22 @@ func (s *PeerPoolSimulationSuite) TearDown() {
 		p.Stop()
 	}
 }
+
+func TestMaxPeersOverflow(t *testing.T) {
```

Comment:
Surely it's tested in here but looking on the test I find nothing about *max peers* nor an *overflow*. Comments describing the story inside the test would help.